script.js

```diff
@@ -0,0 +1,908 @@
+    fadeOutCurrHandAnimation();
+
+    setTimeout(() => {
+      cardsContainer.innerText = '';
+      displayCoverCard();
+    }, 1000);
+
+    setTimeout(() => {
+      coverCardAnimateOut();
+    }, 2500);
+
+    delayInDrawingCardsAnimation = 3500;
+  }
+  shuffledDeck = shuffleCards(makeDeck());
+  setTimeout(() => {
+    cardsContainer.innerText = '';
+
+    // clear playerHand first before drawing initial hand
+    playerHand.length = 0;
+
+    // clear cache of previous' hand's win combi
+    nameOfWinCombi = 'No winning hand.\n Please insert credits to continue';
+    drawInitialHand();
+    checkForWinCombi();
+    console.log('creditsLeft', creditsLeft);
+
+    // Calculate the level of payout and allow relevant pointer to scroll
+    calcPayOutLevel();
+
+    // Scroll the top header to display the winning combi
+    topScoreScrollDisplay();
+
+    // reset insert credits and deduct from credits left:
+    creditsLeft -= numCreditsInserted;
+    numCreditsInserted = 0;
+
+    // reset stats display and make a new one;
+    statsDisplay.innerText = '';
+    createGameStatsDisplay();
+  }, delayInDrawingCardsAnimation);
+
+  gameRound += 1;
+};
+
+// Function that creates a button for dealing cards and run corresponding events
+const createDealCardsBtn = () => {
+  const dealBtn = document.createElement('button');
+  dealBtn.classList.add('btn');
+  dealBtn.classList.add('btn-design5');
+  dealBtn.setAttribute('id', 'dealBtn');
+  dealBtn.innerText = 'DEAL';
+  dealBtn.addEventListener('click', () => {
+    console.log(gameRound, 'gameRound');
+    // reset the top scrolling machine displays and recreate it
+
+    if (numCreditsInserted > 0) {
+      if (gameRound > 0) {
+        refreshScrollMachineDisplay();
+      }
+      dealCardsAndAnimateMsg();
+    } else {
+      statsDisplay.innerText = '';
+      createGameStatsDisplay('Please insert \ncredits to deal!');
+      outputMsgDisplay.classList.add('animate__animated');
+      outputMsgDisplay.classList.add('animate__flash');
+    }
+    gameRound += 1;
+  });
+  buttonsContainer.appendChild(dealBtn);
+};
+
+// Function that creates a button that allows user to swap cards
+const createSwapCardsBtn = () => {
+  const swapBtn = document.createElement('button');
+  swapBtn.classList.add('btn');
+  swapBtn.classList.add('btn-design1');
+  swapBtn.setAttribute('id', 'swapBtn');
+  swapBtn.innerText = 'SWAP';
+
+  swapBtn.addEventListener('click', () => {
+    if (gameState === 'playerNextMove') {
+      gameState = 'gameOver';
+      playerHand.map((currentCard, index) => {
+        if (currentCard.holdStatus === false) {
+          const newCard = shuffledDeck.pop();
+          // const newCard = simulatedHand.pop();
+          playerHand.splice(index, 1, newCard);
+          displayNewDrawnCards(newCard, index);
+        }
+      });
+      checkForWinCombi();
+      refreshScrollMachineDisplay();
+      // calculate score and add to creditsLeft
+      calcHandScore();
```

Comment:
`calcHandScore` function name is misleading because it's not really calculating anything- maybe `updateCreditsLeft` would be a more appropriate function name in this context because the primary purpose of the function is to update num credits left after looking up hand score from a table based on hand rank.

Reply:
Agreed - sometimes as the program gets longer and more complicated, I tend to change the initial purpose of the function, but always get stuck on getting the proper naming in place. Thanks for pointing this out.